### PR TITLE
Setup local cache for AGS calculations

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: workers/src/main/scala/workers/package.scala [hunk 1]
```diff
@@ -40,3 +42,5 @@ object PromiseConverter:
 given PromiseConverter[IO] = new PromiseConverter[IO] {
   inline def convert[A](promise: => js.Promise[A]): IO[A] = IO.fromPromise(IO(promise))
 }
+
+extension (t: IO.type) def now(): IO[Instant] = IO(Instant.now)
```
Comment (rpiaggio) [inline workers/src/main/scala/workers/package.scala:46]: Nice. Maybe we can move this to lucuma-core in the future?

File: workers/src/main/scala/workers/AgsServer.scala [hunk 2]
```diff
@@ -20,18 +25,34 @@ object AgsServer extends WorkerServer[IO, AgsMessage.Request] {
...
+    for
+      self  <- IO(dom.DedicatedWorkerGlobalScope.self)
+      cache <- Cache.withIDB[IO](self.indexedDB.toOption, "ags")
+      _     <- cache.evict(CacheRetention).start
+    yield invocation =>
+      invocation.data match {
+        case req @ AgsMessage.Request(_, _, _, _, _, _, _) =>
+          val cacheableRequest =
+            Cacheable(CacheName("ags"), CacheVersion(AgsCacheVersion), agsCalculation)
+          (IO.now() >>= (now => cache.evict(now.minus(CacheRetention)).start)) >>
```
Comment (rpiaggio) [inline workers/src/main/scala/workers/AgsServer.scala:55]: Do we want to run eviction on every request? We are alrady running it on startup.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: workers/src/main/scala/workers/AgsServer.scala (hunk 2 above)
Comment (cquiroz) [reply workers/src/main/scala/workers/AgsServer.scala:55]: ah you're right. I guess it will be kind of expensive if it gets longer

**Pull request merged**